fog/ingest/server/src/controller.rs

```diff
@@ -1028,11 +1081,34 @@ where
         ingress_public_key: &CompressedRistrettoPublic,
         state: &mut MutexGuard<IngestControllerState>,
     ) -> Result<IngressPublicKeyStatus, Error> {
-        self.update_enclave_report_cache()?;
+        // Get a report and check that it makes sense with what we think is happening
+        let report = {
+            let report = self.enclave.get_ias_report()?;
+            // Check that key in report data matches ingress_public_key.
+            // If not, then there is some kind of race.
+            let found_key = try_extract_ingress_pubkey_from_report(&report)?;
+            if &found_key == ingress_public_key {
```

Comment:
this is some defensive programming:
even if our code doesn't actually have a race here today, that doesn't mean that it won't in the future.

if this condition fails, and we publish a report to the fog db that doesn't actually match the public key, that is likely catastrophic because the users will encrypt for a key which isn't the same as what the enclave is decrypting for, and it will be very complicated to mitigate that after the fact, and possibly require the users to download and scan a very large amount of data. so just double checking and refusing to publish if we can't fix the mismatch seems like the best thing to do.

Reply:
might be nice to just include this as a comment